src/test/scala/examples/DecoupledAdvTester.scala

```diff
@@ -0,0 +1,69 @@
+// See LICENSE for license details.
+
+package examples
+
+import chisel3._
+import chisel3.util.{EnqIO, DeqIO, Queue}
+import chisel3.iotesters.{ChiselFlatSpec, AdvTester}
+
+class SmallOdds5(filter_width: Int) extends Module {
+
+  class FilterIO extends Bundle {
+    val in = DeqIO(UInt(width = filter_width))
+    val out = EnqIO(UInt(width = filter_width))
+  }
+
+  class Filter(isOk: UInt => Bool) extends Module {
+    val io = IO(new FilterIO)
+
+    io.in.ready := io.out.ready
+    io.out.bits := io.in.bits
+
+    io.out.valid := io.out.ready && io.in.valid && isOk(io.in.bits)
+  }
+
+  val io = IO(new FilterIO())
+
+  def buildFilter(): Unit = {
```

Comment:
Does this need to be a method? As far as I can tell there's no re-use and it doesn't increase readability (because it's the only thing being called in the constructor)